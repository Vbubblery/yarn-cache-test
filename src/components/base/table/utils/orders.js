import { frozenPeriods, isFrozen, isMoq, isLotSize } from "./planning";
import { defaultTableValueFormatting, isNotPast } from "./base.js";

export const ORDER_STATUS = {
  VALIDATED: "validated",
  LOCKED: "locked"
};
export const ORDER_FREEZE_STATUS = {
  PARTIAL: "partlyFrozen",
  TOTAL: "fullyFrozen"
};

export const getOrderStatus = status => {
  switch (status) {
    case 2:
      return ORDER_STATUS.VALIDATED;
    case 3:
    case 4:
      return ORDER_STATUS.LOCKED;
    default:
      return undefined;
  }
};

export const getOptionsForOrders = options => {
  const { leadTimes, periods, periodType, productData } = options;
  return {
    periodType,
    periods,
    frozenPeriods: frozenPeriods(
      productData?.leadTime || {},
      periods,
      periodType,
      leadTimes
    ),
    isAggregated: options.isAggregated,
    access: options.access,
    leadTimes: options.leadTimes,
    units: options.productData?.units,
    moq: options.productData?.moq,
    lotSize: options.productData?.lotSize
  };
};

export const getOptionsForStatus3Supplies = options => {
  const { periods, periodType, isAggregated, access } = options;
  return {
    periodType,
    periods,
    isAggregated,
    access,
    units: options.productData?.units
  };
};

export const getOptionsForDemand = options => ({
  periodType: options.periodType,
  periods: options.periods,
  isAggregated: options.isAggregated,
  access: options.access
});

export const getOrdersValues = (orders, options) => {
  if (orders.isParent) {
    return orders.values.map((value, index) => ({
      value,
      isPast: index === 0
    }));
  }

  const {
    isAggregated,
    moq: moqBySupplier,
    lotSize: lotSizeBySupplier,
    ...otherOptions
  } = options;
  const { editedBy, orderStatus, supplier, isParent } = orders;
  return orders.values.map((value, index) => {
    const status = orderStatus && getOrderStatus(orderStatus[index]);
    const isByFlowlity = editedBy && editedBy[index] === "flowlity";
    const frozenStatus = isFrozen(index, supplier, options);
    const moq = isMoq(supplier, isAggregated, orders, moqBySupplier);
    const lotSize = isLotSize(
      supplier,
      isAggregated,
      orders,
      lotSizeBySupplier
    );
    return {
      value,
      isByFlowlity,
      status,
      moq,
      lotSize,
      isPast: index === 0,
      tooltip: supplier !== null || supplier !== undefined,
      isEditable: options.access && !isAggregated && !isParent && index > 0,
      ...(!isNaN(value) && { isFrozen: frozenStatus }),
      ...otherOptions
    };
  });
};

export const getStatus3SuppliesValues = (supplies, options) => {
  const { isAggregated, access, ...otherOptions } = options;
  return supplies.values.map((value, index) => {
    return {
      value,
      isPast: index === 0,
      isEditable: access && !isAggregated && index > 0,
      ...otherOptions
    };
  });
};

export const getAllOrdersValues = (allOrders, options) => {
  return allOrders.values.map((value, index) => {
    return {
      value,
      isPast: !isNotPast(index, options.periods, options.periodType)
    };
  });
};

export const allOrdersCalc = (value, isPast) => {
  if (isPast) return null;
  else if (!value) return 0;
  return defaultTableValueFormatting(value);
};
