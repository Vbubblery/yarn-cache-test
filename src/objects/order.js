import { validate } from "jsonschema";
import OrderSchema, { orderSupplySchema } from "@/schemas/order";

export class OrderSupply {
  constructor(model = {}) {
    this._model = model;
    this._id = model.id;
    this._productId = model.product_site;
    this._externalId = model.external_id;
    this._isDefaultUnit = model.is_default_unit;
    this._defaultUnitValue = model.default_unit_value;
    this._quantity = model.quantity;
    this._secondaryUnitValue = model.second_unit_value;
    this._secondaryUnitQuantity = model.second_unit_quantity;
    this._orderId = model.order;
    this._name = model.name;
    this._unit = model.unit;
    this._secondaryUnit = model.second_unit;
    this._leadTime = model.lead_time;
    this._price = model.product_price;
    this._totalPrice = model.total_price;
    this._moq = model.moq;
    this._description = model.description;
    this._lotSize = model.lot_size;
    this._stockAtEndOfPeriod = model.stock_at_end_of_period;
    this._minStockAtEndOfPeriod = model.minimum_stock_at_end_of_period;
    this._totalDemand = model.total_demand;
  }
  static create(model) {
    return validate(model, OrderSupply.schema())
      ? new OrderSupply(model)
      : null;
  }

  static schema() {
    return orderSupplySchema;
  }

  get externalId() {
    return this._externalId;
  }

  get defaultUnitValue() {
    return this._defaultUnitValue;
  }

  get secondaryUnitValue() {
    return this._secondaryUnitValue;
  }

  get secondaryUnitQuantity() {
    return this._secondaryUnitQuantity;
  }

  get secondaryUnit() {
    return this._secondaryUnit;
  }

  get moq() {
    return this._moq;
  }

  get description() {
    return this._description;
  }

  get lotSize() {
    return this._lotSize;
  }

  get id() {
    return this._id;
  }

  get productId() {
    return this._productId;
  }

  get quantity() {
    return this._quantity;
  }

  get orderId() {
    return this._orderId;
  }

  get name() {
    return this._name;
  }

  get unit() {
    return this._unit;
  }

  get leadTime() {
    return this._leadTime;
  }

  get price() {
    return this._price;
  }

  get totalPrice() {
    return this._totalPrice;
  }

  get isDefaultUnit() {
    return this._isDefaultUnit;
  }

  get minStockAtEndOfPeriod() {
    return this._minStockAtEndOfPeriod;
  }
  get stockAtEndOfPeriod() {
    return this._stockAtEndOfPeriod;
  }
  get totalDemand() {
    return this._totalDemand;
  }
  set id(id) {
    this._id = id;
  }

  set productId(productId) {
    this._productId = productId;
  }

  set orderId(orderId) {
    this._orderId = orderId;
  }

  set quantity(quantity) {
    this._quantity = quantity;
  }

  set name(name) {
    this._name = name;
  }

  set unit(unit) {
    this._unit = unit;
  }

  set leadTime(leadTime) {
    this._leadTime = leadTime;
  }
  set totalPrice(totalPrice) {
    this._totalPrice = totalPrice;
  }

  add(otherSupply) {
    return new OrderSupply({
      ...this._model,
      quantity: this.quantity + otherSupply.quantity
    });
  }
}

export class Order {
  constructor(model = {}) {
    this._customerSiteId = model.customer_site_id;
    this._customerSiteName = model.customer_site_name;
    this._date = model.date;
    this._externalId = model.external_id;
    this._fullTruckQuantity = model.full_truck_quantity;
    this._fullTruckUnit = model.full_truck_unit;
    this._id = model.id;
    this._minOrderQuantity = model.minimum_order_quantity;
    this._minOrderUnit = model.minimum_order_unit;
    this._orderFrequency = model.order_frequency;
    this._previousOrderDate = model.previous_order_date;
    this._sendBefore = model.send_before;
    this._shipBefore = model.ship_before;
    this._statusId = model.status_id;
    this._supplierId = model.supplier_id;
    this._supplies = (model.supplies || []).map(supply =>
      OrderSupply.create(supply)
    );
    this._supplierName = model.name;
    this._tagGrouper = model.tag_grouper;
    this._leadTime = model.lead_time;
    this._lastSftpExportDate = model.last_sftp_export_date;
  }
  static create(model) {
    return validate(model, Order.schema()) ? new Order(model) : null;
  }
  static schema() {
    return OrderSchema;
  }

  // Getters
  get customerSiteId() {
    return this._customerSiteId;
  }
  get customerSiteName() {
    return this._customerSiteName;
  }
  get orderFrequency() {
    return this._orderFrequency;
  }
  get previousOrderDate() {
    return this._previousOrderDate;
  }
  get sendBefore() {
    return this._sendBefore;
  }
  get shipBefore() {
    return this._shipBefore;
  }
  get id() {
    return this._id;
  }
  get statusId() {
    return this._statusId;
  }
  get date() {
    return this._date;
  }
  get supplierId() {
    return this._supplierId;
  }
  get supplies() {
    return this._supplies;
  }
  get fullTruckQuantity() {
    return this._fullTruckQuantity;
  }
  get fullTruckUnit() {
    return this._fullTruckUnit;
  }
  get minOrderQuantity() {
    return this._minOrderQuantity;
  }
  get minOrderUnit() {
    return this._minOrderUnit;
  }
  get supplierName() {
    return this._supplierName;
  }
  get externalId() {
    return this._externalId;
  }
  get tagGrouper() {
    return this._tagGrouper;
  }
  get leadTime() {
    return this._leadTime;
  }
  get lastSftpExportDate() {
    return this._lastSftpExportDate;
  }

  // Setters
  set id(id) {
    this._id = id;
  }

  set statusId(statusId) {
    this._statusId = statusId;
  }
  set date(date) {
    this._date = date;
  }
  set supplierId(supplierId) {
    throw new Error(
      "Cannot change a supplier for an existing order. Create a new order instead"
    );
  }
  set supplies(supplies) {
    this._supplies = supplies;
  }
  set fullTruckQuantity(fullTruckQuantity) {
    this._fullTruckQuantity = fullTruckQuantity;
  }
  set fullTruckUnit(fullTruckUnit) {
    this._fullTruckUnit = fullTruckUnit;
  }
  set minOrderQuantity(minOrderQuantity) {
    this._minOrderQuantity = minOrderQuantity;
  }
  set minOrderUnit(minOrderUnit) {
    this._minOrderUnit = minOrderUnit;
  }
  set supplierName(supplierName) {
    this._supplierName = supplierName;
  }
  set externalId(externalId) {
    this._externalId = externalId;
  }
  set tagGrouper(tagGrouper) {
    this._tagGrouper = tagGrouper;
  }
  set lastSftpExportDate(lastSftpExportDate) {
    this._lastSftpExportDate = lastSftpExportDate;
  }
}
