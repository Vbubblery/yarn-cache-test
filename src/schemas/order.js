export default {
  type: "object",
  properties: {
    id: {
      type: "integer",
      minimum: 1
    },
    status_id: {
      type: "integer",
      minimum: 1,
      maximum: 4
    },
    supplier_id: {
      type: "integer",
      minimum: 1
    },
    date: {
      type: "string"
    },
    supplies: {
      type: "object"
    }
  },
  required: ["id", "status_id", "supplier_id", "date", "supplies"]
};

export const orderSupplySchema = {
  type: "object",
  properties: {
    id: {
      type: "integer",
      minimum: 1
    },
    product_site: {
      type: "integer",
      minimum: 1
    },
    quantity: {
      type: "integer"
    },
    order: {
      type: "integer",
      minimum: 1
    },
    name: {
      type: "string"
    }
  },
  required: ["id", "product_site", "quantity", "name"]
};
