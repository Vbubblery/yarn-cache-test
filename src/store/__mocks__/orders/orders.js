import ordersData from "./modules/ordersData";
import ordersDataOptions from "./modules/ordersDataOptions";

const namespaced = true;

const modules = {
  ordersData,
  ordersDataOptions
};

export default {
  namespaced,
  modules
};
