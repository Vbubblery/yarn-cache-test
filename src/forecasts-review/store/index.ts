import { computed, SetupContext, ComputedRef } from "@vue/composition-api";
import { PLANNING_DEMAND_VIEW } from "@/forecasts-review/typings";
import { Site } from "@/objects/site";

export const useBaseResources = (context: SetupContext) => {
  const store = context.parent?.$store || null;

  const getMe = computed<{ locale: string }>(
    () => store?.getters["common/getMe"]
  );
  const planningDemandView: ComputedRef<PLANNING_DEMAND_VIEW> = computed(
    () => store?.getters["common/getDefaultView"]
  );
  const currentSite = computed<Site>(
    () => store?.getters["common/getCurrentSite"]
  );
  const tablesData = computed(() => store?.getters["demand/getTablesData"]);
  const demandGetChartData = computed(
    () => store?.getters["demand/getChartData"]
  );
  const demandGetIsLoadingChart = computed(
    () => store?.getters["demand/getIsLoadingChart"]
  );
  const demandGetIsLoadingTables = computed(
    () => store?.getters["demand/getIsLoadingTables"]
  );
  const planningGetChartData = computed(
    () => store?.getters["planning/getChartData"]
  );
  const planningGetIsLoadingChart = computed(
    () => store?.getters["planning/getIsLoadingChart"]
  );
  const planningGetIsLoadingTables = computed(
    () => store?.getters["planning/getIsLoadingTables"]
  );
  const selectedTagIds = computed(
    () => store?.getters["common/getSelectedTags"]
  );
  const selectedCustomerId = computed(() =>
    currentSite.value.isSupplierView
      ? store?.getters["demand/getSelectedCustomerSiteId"]
      : null
  );
  const updateDefaultView = (view: PLANNING_DEMAND_VIEW) => {
    store?.commit("common/updateDefaultView", view);
  };
  const updateSelectedProducts = (items: object[]) => {
    store?.dispatch("common/updateSelectedProducts", items);
  };
  const updateSelectedTags = (items: object[]) => {
    store?.dispatch("common/updateSelectedTags", items);
  };
  const updateDataType = (itemType: string) => {
    store?.commit("common/updateDataType", itemType);
  };
  const resetDemandState = () => {
    store?.commit("demand/resetState");
  };
  const updateDemandPage = (pageNum: number) => {
    store?.commit("demand/updatePage", pageNum);
  };
  const updateDemandTableOnly = (onlyTable: boolean) => {
    store?.commit("demand/updateOnlyTable", onlyTable);
  };
  const fetchData = (
    selectedView: undefined | PLANNING_DEMAND_VIEW = undefined
  ) => {
    const currentView = selectedView ?? planningDemandView.value;
    store?.dispatch(
      currentView === PLANNING_DEMAND_VIEW.DEMAND
        ? "demand/fetchDemandData"
        : "planning/updatePlanningData"
    );
  };
  const updateSelectedCustomerSiteId = (customerSiteId: number | null) => {
    store?.commit("demand/updateSelectedCustomerSiteId", customerSiteId);
  };
  return {
    getMe,
    planningDemandView,
    updateDefaultView,
    currentSite,
    tablesData,
    demandGetChartData,
    demandGetIsLoadingChart,
    demandGetIsLoadingTables,
    planningGetChartData,
    planningGetIsLoadingChart,
    planningGetIsLoadingTables,
    selectedCustomerId,
    selectedTagIds,
    updateSelectedProducts,
    updateSelectedTags,
    updateDataType,
    resetDemandState,
    updateDemandPage,
    updateDemandTableOnly,
    fetchData,
    updateSelectedCustomerSiteId
  };
};
