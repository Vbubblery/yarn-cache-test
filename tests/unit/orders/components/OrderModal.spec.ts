import { renderComponent } from "tests/test-utils";
import OrderModal from "@/orders/components/OrderModal.vue";
import { Order } from "@/objects/order";
// eslint-disable-next-line jest/no-mocks-import
import { store } from "@/store/__mocks__/store";
import Vue from "vue";
import { Modal } from "ant-design-vue";
import { waitFor, RenderOptions, fireEvent } from "@testing-library/vue";
// eslint-disable-next-line import/no-extraneous-dependencies
import Swal from "sweetalert2";
import * as OrdersBackend from "@/orders/backend";
jest.mock("ant-design-vue", () => ({
  ...jest.requireActual("ant-design-vue"),
  DatePicker: jest.fn()
}));

const fetchOrder = jest.spyOn(OrdersBackend, "getOrderByFilters");
const spyOrdersByStatus = jest.spyOn(OrdersBackend, "getOrdersCountByStatus");
jest.spyOn(OrdersBackend, "fetchOrdersCountByStatus").mockImplementation();
jest.spyOn(OrdersBackend, "updateOrderStatus").mockImplementation();
jest.spyOn(Swal, "fire").mockResolvedValue({
  isConfirmed: true,
  isDenied: false,
  isDismissed: false
});
// const deleteOrderSpy = jest
//   .spyOn(OrdersBackend, "deleteBackendOrder")
//   .mockImplementation();
// Reduce the modal rendering boilerplate
const renderModal = async (opts: RenderOptions<Vue>) => {
  const renderResult = renderComponent(OrderModal, opts);
  await waitFor(() => {}); // Must wait until the modal is mounted
  return renderResult;
};

const basicProps = {
  visible: true,
  orderProp: new Order(),
  dateLabelToDisplay: "",
  shouldDisplayActions: true,
  availableActions: []
};

describe("OrderModal", () => {
  beforeAll(() => Vue.use(Modal));
  beforeEach(() => {
    spyOrdersByStatus.mockClear();
    fetchOrder.mockClear();
  });
  it("Renders the component", async () => {
    const { baseElement } = await renderModal({
      props: basicProps,
      store
    });
    /**
     * Checking the snapshot of the baseElement instead of html
     * because the modal is not mounted in its parent container
     * but in a container generated in the body
     */
    expect(baseElement).toMatchSnapshot();
  });
  it("Does not show edit button if order status not < 2", async () => {
    const { getByTestId } = await renderModal({
      props: { ...basicProps, orderProp: new Order({ status_id: 2 }) },
      store
    });
    expect(() => getByTestId(/order_modal_edit_button/)).toThrow();
  });
  it("Should be in edit mode by default if status is 1", async () => {
    const { getByTestId } = await renderModal({
      props: {
        ...basicProps,
        orderProp: new Order({
          status_id: 1,
          date: null,
          supplies: [{ id: 1, quantity: 1, moq: 2 }]
        })
      },
      stubs: {
        DatePicker: {
          template: "<span />"
        }
      },
      store
    });
    const productInput = getByTestId(/order_modal_product_input/);
    await fireEvent.update(productInput, "5");
    expect(() => getByTestId(/order_modal_save_button/)).toBeDefined();
  });
  // TODO: Check if a cancel button appears after click on edit
  it.todo("Gives the possibility to cancel when a change is made");
  // TODO: Click on edit, input new values in supplies, cancel, then check if the values displayed = the values in the orderProp
  it.todo("Switches back to initial values when cancelling change");
  // TODO: Check if backend service is called with the right arguments after click on confirm
  it.todo("Saves changes when approving changes");
  it.todo("Updates the days of coverage on supply quantity input");
  it("should display confirm button for purchase requisition order", async () => {
    const newOrder = new Order({
      id: 12,
      status_id: 2,
      date: "2022-05-30",
      supplies: [{ quantity: 1 }]
    });
    fetchOrder.mockResolvedValue(newOrder);
    const { getByTestId, updateProps } = await renderModal({
      props: {
        ...basicProps,
        orderProp: new Order({
          id: 12,
          status_id: 1,
          date: "2022-05-30",
          supplies: [{ quantity: 1 }]
        }),
        availableActions: ["delete"]
      },
      store
    });
    const confirmButton = getByTestId(/order_modal_confirm_button/);
    expect(confirmButton).toBeDefined();
    await waitFor(async () => {
      await fireEvent.click(confirmButton);
      expect(fetchOrder).toHaveBeenCalled();
    });
    await waitFor(async () => {
      await updateProps({ ...basicProps, orderProp: newOrder });
      const cancelButton = getByTestId(/order_modal_cancel_button/);
      expect(cancelButton).toBeDefined();
    });
  });
  it("should display discard button for confirmed purchase requisition order", async () => {
    jest.spyOn(OrdersBackend, "deleteBackendOrder").mockImplementation();
    const newOrder = new Order({
      id: 12,
      status_id: 1,
      date: "2022-05-30",
      supplies: [{ quantity: 1 }]
    });
    fetchOrder.mockResolvedValue(newOrder);
    const { getByTestId, updateProps } = await renderModal({
      props: {
        ...basicProps,
        orderProp: new Order({
          id: 12,
          status_id: 2,
          date: "2022-05-30",
          supplies: [{ quantity: 1 }]
        }),
        availableActions: ["delete"]
      },
      store
    });

    const cancelButton = getByTestId(/order_modal_cancel_button/);
    expect(cancelButton).toBeDefined();
    await waitFor(async () => {
      await fireEvent.click(cancelButton);
      expect(fetchOrder).toHaveBeenCalled();
    });
    await waitFor(async () => {
      await updateProps({ ...basicProps, orderProp: newOrder });
      const confirmButton = getByTestId(/order_modal_confirm_button/);
      expect(confirmButton).toBeDefined();
    });
  });
  it("shouldn't display confirm and discard button for status3 orders", async () => {
    jest.spyOn(OrdersBackend, "deleteBackendOrder").mockImplementation();
    const { findByTestId } = await renderModal({
      props: {
        ...basicProps,
        orderProp: new Order({
          id: 12,
          status_id: 3,
          supplies: [{ quantity: 1 }]
        })
      },
      store
    });
    await expect(
      findByTestId(/order_modal_cancel_button/)
    ).rejects.toBeTruthy();
    await expect(
      findByTestId(/order_modal_confirm_button/)
    ).rejects.toBeTruthy();
  });
  it("shouldn't display confirm and discard button for status4 orders", async () => {
    const { findByTestId } = await renderModal({
      props: {
        ...basicProps,
        orderProp: new Order({
          id: 12,
          status_id: 4,
          supplies: [{ quantity: 1 }]
        }),
        availableActions: ["delete"]
      },
      store
    });
    await expect(
      findByTestId(/order_modal_cancel_button/)
    ).rejects.toBeTruthy();
    await expect(
      findByTestId(/order_modal_confirm_button/)
    ).rejects.toBeTruthy();
  });
  // eslint-disable-next-line jest/no-commented-out-tests
  // it("Delete order", async () => {
  //   const { getByTestId } = await renderModal({
  //     props: { ...basicProps, availableActions: ["delete"] },
  //     store
  //   });
  //   await fireEvent.click(getByTestId(/delete_button_order_details/));
  //   expect(deleteOrderSpy).toHaveBeenCalled();
  // });
  it("Opens the planning modal when clicking on Planning button", async () => {
    const { getByTestId } = await renderModal({
      props: {
        ...basicProps,
        orderProp: new Order({
          status_id: 1,
          supplies: [{ quantity: 2 }]
        })
      },
      store
    });
    await fireEvent.click(getByTestId(/order_modal_open_planning_button/));
    await waitFor(() =>
      expect(() => getByTestId(/order_modal_planning/)).not.toThrow()
    );
  });
});
