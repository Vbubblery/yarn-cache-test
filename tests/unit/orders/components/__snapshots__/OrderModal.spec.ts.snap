// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderModal Renders the component 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div>
    <div
      class="ant-modal-root"
    >
      <transition-stub
        appear="true"
      >
        <div
          class="ant-modal-mask"
        />
      </transition-stub>
      <div
        class="ant-modal-wrap ant-modal-centered"
        role="dialog"
        tabindex="-1"
      >
        <transition-stub
          appear="true"
        >
          <div
            class="ant-modal pb-0 h-full"
            role="document"
            style="width: 100%; height: 100%;"
          >
            <div
              aria-hidden="true"
              style="width: 0px; height: 0px; overflow: hidden;"
              tabindex="0"
            />
            <div
              class="ant-modal-content"
            >
              <div
                class="ant-modal-body"
                style="padding: 0px;"
              >
                <!---->
                 
                <div
                  class="flex flex-col h-screen bg-white font-montserrat"
                  data-testid="order_modal_container"
                >
                  <div
                    class="flex justify-between items-center px-10 py-5 col-span-10 row-span-1 border-b-2"
                  >
                    <div
                      class="flex items-center"
                    >
                      <button
                        class="mr-4 text-gray-400 btn btn--secondary relative h-8 flex justify-center items-center rounded-md cursor-pointer w-8 h-8"
                        type="button"
                      >
                        <svg
                          class="btn__icon"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <use
                            href="/assets/svg/sprite.svg#svg-arrow-left-s-line"
                            xlink:href="/assets/svg/sprite.svg#svg-arrow-left-s-line"
                          />
                        </svg>
                         
                        <!---->
                      </button>
                       
                      <!---->
                       
                      <div
                        class="flex flex-col justify-center font-roboto"
                      >
                        <div
                          class="mb-0.5 text-[16px] leading-[19px] text-black"
                        >
                          <span>
                            
          orders.orderModalTitle
        
                          </span>
                           
                          <span
                            class="uppercase text-green-700"
                          >
                            
          
        
                          </span>
                        </div>
                         
                        <div
                          class="text-[14px] leading-[16px] text-gray-600"
                        >
                          
        orders.orderId: 
      
                        </div>
                      </div>
                    </div>
                     
                    <div
                      class="flex items-center"
                    >
                      <!---->
                       
                      <!---->
                       
                      <!---->
                       
                      <!---->
                    </div>
                  </div>
                   
                  <div
                    class="flex justify-between flex-1 overflow-hidden"
                  >
                    <div
                      class="flex-1 flex flex-col overflow-hidden"
                    >
                      <div
                        class="grid pt-6 grid-cols-24 text-sm text-fl-dark-grey"
                      >
                        <div
                          class="col-start-2 p-2 pl-0 col-span-8"
                        >
                          
      orders.reference
    
                        </div>
                         
                        <div
                          class="col-span-3 p-2 text-right"
                        >
                          
      orders.quantity
    
                        </div>
                         
                        <div
                          class="col-span-3 p-2 text-right"
                        >
                          
      orders.daysOfCoverage
    
                        </div>
                         
                        <div
                          class="col-span-2 p-2 text-right"
                        >
                          
      orders.lotSize
    
                        </div>
                         
                        <div
                          class="col-span-2 p-2 text-right"
                        >
                          
      orders.moq
    
                        </div>
                         
                        <div
                          class="col-span-3 p-2 pr-8 text-right"
                        >
                          
      orders.value
    
                        </div>
                      </div>
                       
                      <div
                        class="flex-1 overflow-auto pt-[1px] pb-1"
                      />
                    </div>
                     
                    <div
                      class="min-w-[436px] max-w-[436px] flex flex-col justify-between h-full px-10 py-8 overflow-auto bg-gray-100 border-gray-300 border-l"
                    >
                      <div>
                        <p
                          class="text-xl"
                        >
                          <span
                            class="text-gray-900"
                          >
                            
        orders.orderTitleStatusundefined
      
                          </span>
                           
                          <span>
                            
        - orders.orderTitleDetail
      
                          </span>
                        </p>
                         
                        <div
                          class="mt-1 min-h-[42px]"
                        >
                          <p
                            class="flex items-center"
                          >
                            <svg
                              class="mr-1 text-green-600"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <use
                                href="/assets/svg/sprite.svg#svg-checkbox-circle-line"
                                xlink:href="/assets/svg/sprite.svg#svg-checkbox-circle-line"
                              />
                            </svg>
                             
                            <span
                              class="text-green-700"
                              data-testid="order_modal_met_constraints"
                            >
                              
          orders.constraintsMetMessage
        
                            </span>
                          </p>
                        </div>
                         
                        <div
                          class="my-12 space-y-4"
                        >
                          <div
                            class="grid grid-cols-2"
                          >
                            <div
                              class="col-span-1 text-sm flex items-center"
                            >
                              
          orders.deliveryDate
        
                            </div>
                             
                            <div
                              class="col-span-1 text-black"
                            >
                              
          Invalid Date
        
                            </div>
                          </div>
                           
                          <div
                            class="grid grid-cols-2"
                          >
                            <div
                              class="col-span-1 text-sm"
                            >
                              
          orders.latestOrderDate
        
                            </div>
                             
                            <div
                              class="col-span-1 text-black"
                            >
                              
          
        
                            </div>
                          </div>
                           
                          <div
                            class="grid grid-cols-2"
                          >
                            <div
                              class="col-span-1 text-sm"
                            >
                              orders.totalQuantity
                            </div>
                             
                            <div
                              class="col-span-1"
                            >
                              <span
                                class="text-black"
                              >
                                
            0
          
                              </span>
                               
                              <span
                                class="text-xs"
                              >
                                
            
          
                              </span>
                            </div>
                          </div>
                           
                          <div
                            class="grid grid-cols-2"
                          >
                            <div
                              class="col-span-1 text-sm"
                            >
                              orders.addedItems
                            </div>
                             
                            <div
                              class="col-span-1"
                            >
                              <span
                                class="text-black"
                              >
                                
            0
          
                              </span>
                               
                              <span
                                class="text-xs"
                              >
                                
            orders.items
          
                              </span>
                            </div>
                          </div>
                           
                          <div
                            class="grid grid-cols-2 mb-8"
                          >
                            <div
                              class="col-span-1 text-sm"
                            >
                              orders.groupedBy
                            </div>
                             
                            <div
                              class="col-span-1"
                            >
                              
          -
        
                            </div>
                          </div>
                        </div>
                         
                        <div
                          class="grid grid-cols-2 mb-8"
                        >
                          <div
                            class="col-span-1 text-sm"
                          >
                            orders.inTotal
                          </div>
                           
                          <div
                            class="col-span-1 text-base"
                          >
                            <span
                              class="text-green-700"
                              data-testid="order_modal_total_price"
                            >
                              
          0,00
        
                            </span>
                             
                            <span
                              class="text-xs"
                            >
                              €
                            </span>
                          </div>
                        </div>
                      </div>
                       
                      <div
                        class="mt-auto space-y-4"
                      >
                        <div
                          class="text-xl"
                        >
                          orders.orderConstraints
                        </div>
                         
                        <div
                          class="mt-4 grid grid-cols-2"
                        >
                          <div
                            class="col-span-1 text-sm flex items-center relative"
                          >
                            <!---->
                            
        orders.orderFrequency
      
                          </div>
                           
                          <div
                            class="col-span-1"
                          >
                            <span
                              class="text-black"
                            >
                              
          -
        
                            </span>
                             
                            <span
                              class="text-xs"
                            >
                              
          
        
                            </span>
                          </div>
                        </div>
                         
                        <div
                          class="grid grid-cols-2"
                        >
                          <div
                            class="col-span-1 text-sm flex items-center relative"
                          >
                            <!---->
                            
        orders.fullTruckQty
      
                          </div>
                           
                          <div
                            class="col-span-1"
                          >
                            <span
                              class="text-black"
                            >
                              
          -
        
                            </span>
                             
                            <span
                              class="text-xs"
                            >
                              
                            </span>
                          </div>
                        </div>
                         
                        <div
                          class="grid grid-cols-2"
                        >
                          <div
                            class="col-span-1 text-sm flex items-center relative"
                          >
                            <!---->
                            
        orders.moq
      
                          </div>
                           
                          <div
                            class="col-span-1"
                          >
                            <span
                              class="text-black"
                            >
                              
          -
        
                            </span>
                             
                            <span
                              class="text-xs"
                            >
                              
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              style="width: 0px; height: 0px; overflow: hidden;"
              tabindex="0"
            />
          </div>
        </transition-stub>
      </div>
    </div>
  </div>
</body>
`;
